src/main/java/org/gaul/s3proxy/S3ProxyHandler.java

```diff
@@ -156,7 +156,8 @@ public void handle(String target, Request baseRequest,
 
         if (identity != null && !hasDateHeader && !hasXAmzDateHeader &&
                 request.getParameter("Expires") == null) {
-            sendSimpleErrorResponse(response, S3ErrorCode.ACCESS_DENIED);
+            sendSimpleErrorResponse(response, S3ErrorCode.ACCESS_DENIED,
+                    Optional.of("No date header provided."));
```

Comment:
Can you mimic AWS more closely?  Running `s3tests.functional.test_headers:test_bucket_create_bad_date_none` reveals this reason:

```
<Error>
    <Code>AccessDenied</Code>
    <Message>AWS authentication requires a valid Date or x-amz-date header</Message>
    <RequestId>4F8BF8887D829814</RequestId>
    <HostId>i4rhOCLqQEJzmbKClcnKkGlQlKBOxYTa6Dv+T/JgvZoK90ESYQjhGivPU4VchK9I</HostId>
</Error>
```

Could you also add this check to s3-tests and submit a pull request?